pkgs/development/compilers/temurin-bin/generate-sources.py

```diff
@@ -6,7 +6,12 @@
 import requests
 import sys
 
+# Enter a GitHub token here to avoid rate limits - use an unprivileged token just in
+# case you accidentally commit it ;). TODO make this a parameter.
+GITHUB_TOKEN = None
```

Comment:
You could also read `GITHUB_TOKEN` from the environment, like nixpkgs-review does

Reply:
I think that won't work because this script is executed in a `nix-shell --pure` `#!`

Reply:
Oh, right. I think you could drop the `--pure`, since it is just a Python script..

Reply:
Hmm, it's kind of nice to have it isolated though. I've made it a parameter as promised in the comment